convert.cc

```diff
@@ -54,18 +54,29 @@ int main(int argc, char* argv[]){
 
 
 	DB_ * x = new Libdb();
-	x->connect_database(src);
+	if (!x->connect_database(src)) {
+	// error is printed in the connect_database function
+	delete x;
+	return 1;
+	}
 	DB_ *b;
 	if (lmdb)
 		b = new LMDB_();
 	else
 		b = new GDBM_();
-	b->create_database(dst);
+	if (!b->create_database(dst)) {
+		std::cerr<<"Failed to create destination database\n";
+		delete x;
+		delete b;
+		return 1;
+	}
 	if (!b->fill_database(x)){
 		std::cerr<<"database filling failed\n";
+		delete x;
+		delete b;
```

Comment:
I think you should close the database first

Reply:
Yes I missed that, I added the missing `close_db` calls and also implemented the function in LibDB_ class to properly close both the cursor and the db. (see the new commit)

Reply:
Although maybe it would be nicer to close the DBs in destructors.

Reply:
Yeah, I think it's best to call the `close_db()` from the destructor

Reply:
Done